.github/workflows/build-and-test-differential.yaml

```diff
@@ -18,9 +23,12 @@ jobs:
             build-depends-repos: build_depends.repos
     steps:
       - name: Check out repository
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
         with:
-          fetch-depth: 0
+          fetch-depth: 1
```

Comment:
It looks like there was an issue with the action not being able to detect updated packages properly if fetch-depth is set to 1. https://github.com/autowarefoundation/autoware.universe/pull/7408#issuecomment-2160242617.

Reply:
- https://github.com/autowarefoundation/autoware.universe/pull/7468
is merged, and this section is updated with https://github.com/autowarefoundation/autoware_common/pull/250/commits/368b4bd4494719ec3f6d6d1d21799360c2c833d5